pkg/types/vsphere/validation/platform.go

```diff
@@ -94,3 +100,18 @@ func validateFolder(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
 
 	return allErrs
 }
+
+// ValidateVCenterAddress validates vCenter to only contain hostname and port
+func ValidateVCenterAddress(vCenter string) error {
+	if strings.TrimSpace(vCenter) == "" {
+		return errors.New("must specify the name of the vCenter")
+	}
+	vCenterURL, err := url.Parse(vCenter)
+	if err != nil {
+		return err
+	}
+	if vCenterURL.RawQuery != "" {
```

Comment:
we need to validate that this does not contain a transport, like https://

Reply:
SplitHostPort just splits a string based on the occurence of first ":". url.Parse(urlString) parses a string and separates transport scheme from domain based on the first occurence of ":".